### ISSUE TITLE
feat: New size property in the ion-chip component

### ISSUE THREAD

**Issue by ramoncarreras** [ionic-team/ionic#18338]
# Feature Request

**Ionic version:**
[x] **4.x**

**Describe the Feature Request**
It would be interesting to have the '_size_' property in the **ion-chip** component, just like ion-button component.

**Related Code**
```typescript
<ion-chip size="large">
  <ion-icon name="pin"></ion-icon>
  <ion-label>Large</ion-label>
</ion-chip>

<ion-chip>
  <ion-icon name="pin"></ion-icon>
  <ion-label>Default</ion-label>
</ion-chip>

<ion-chip size="small">
  <ion-icon name="pin"></ion-icon>
  <ion-label>Small</ion-label>
</ion-chip>
```

**Comment by tallcoder** [comment]
This would be super helpful.

**Comment by numerized** [comment]
YESSS

**Comment by juanbrein** [comment]
PLEASSSEEE

**Comment by Chriserus** [comment]
Can be usefull. Came here looking for answers on how to change chip size :)

**Comment by xc-mezcal** [comment]
Yup! Same here. Is it possible to fix the length of chip to 100% width long?

**Comment by brandyscarney** [comment]
Thanks for the issue! How would you expect a small/large chip to look? I'm able to customize it using CSS, for example: https://codepen.io/brandyscarney/pen/mdejzwo

I've been going through the [Material Design docs](https://material.io/components/chips) and I don't see any places where the size changes. I just want to make sure I understand the use case here. This can currently be done with CSS - adding a size property would actually be more restrictive.

**Comment by juanbrein** [comment]
> Thanks for the issue! How would you expect a small/large chip to look? I'm able to customize it using CSS, for example: https://codepen.io/brandyscarney/pen/mdejzwo
> 
> I've been going through the [Material Design docs](https://material.io/components/chips) and I don't see any places where the size changes. I just want to make sure I understand the use case here. This can currently be done with CSS - adding a size property would actually be more restrictive.

This solution doesn't change the height of the chip in the smaller size

**Comment by brandyscarney** [comment]
Hello @juanbrein! This issue likely started when we added support for [dynamic font scaling](https://ionicframework.com/docs/layout/dynamic-font-scaling#using-dynamic-font-scaling), as components now use `min-height` to allow them to grow in height when the font size increases.

The example I shared earlier can be updated to the following to adjust the small size:

```css
[size="small"] {
  min-height: 20px;
  font-size: 12px;
  padding: 0;
}
```

You can either remove the padding line or update it to a value of your choice.
